connection.go

```diff
@@ -64,14 +64,23 @@ func (mc *mysqlConn) handleParams() (err error) {
 				return
 			}
 
-		// Other system vars
+		// Other system vars accumulated in a single SET command
 		default:
-			params = append(params, param+"="+val)
+			if cmdSet == nil {
+				// Heuristic: 29 chars for each other key=value to reduce reallocations
+				cmdSet = make([]byte, 0, 4+len(param)+1+len(val)+30*(len(mc.cfg.Params)-1))
```

Comment:
Wouldn't it make sense to use either a const-sized buffer on the heap for the common case or try to use the `mc.buf`?

Reply:
A better memory saving would occur by building the `SET NAMES` and `SET` commands just once for all connections as the DSN is frozen. But this is more profound refactoring.

Reply:
#1115 is the first step to move the connection-independent parts of `handleParams` to a `connector` constructor.